Tensile/Source/client/source/ReferenceValidator.cpp

```diff
@@ -122,104 +145,119 @@ namespace Tensile
-                    auto const& typedResult = dynamic_cast<TypedContractionInputs<std::complex<double>> const&>(*inputs);
-                    validateTyped(typedReference, typedResult);
-                }
-                else if(m_problem.a().dataType() == DataType::Half
-                     && m_problem.b().dataType() == DataType::Half
-                     && m_problem.c().dataType() == DataType::Half
-                     && m_problem.d().dataType() == DataType::Half)
-                {
-                    auto const& typedReference = dynamic_cast<TypedContractionInputs<Half> const&>(*m_referenceInputs);
-                    auto const& typedResult = dynamic_cast<TypedContractionInputs<Half> const&>(*inputs);
-                    validateTyped(typedReference, typedResult);
-                }
-                else if(m_problem.a().dataType() == DataType::Int8x4
-                     && m_problem.b().dataType() == DataType::Int8x4
-                     && m_problem.c().dataType() == DataType::Int32
-                     && m_problem.d().dataType() == DataType::Int32)
-                {
-                    auto const& typedReference = dynamic_cast<TypedContractionInputs<Int8x4, Int8x4, int32_t, int32_t> const&>(*m_referenceInputs);
-                    auto const& typedResult = dynamic_cast<TypedContractionInputs<Int8x4, Int8x4, int32_t, int32_t> const&>(*inputs);
-                    validateTyped(typedReference, typedResult);
-                }
-                else if(m_problem.a().dataType() == DataType::Int32
-                     && m_problem.b().dataType() == DataType::Int32
-                     && m_problem.c().dataType() == DataType::Int32
-                     && m_problem.d().dataType() == DataType::Int32)
-                {
-                    auto const& typedReference = dynamic_cast<TypedContractionInputs<int32_t> const&>(*m_referenceInputs);
-                    auto const& typedResult = dynamic_cast<TypedContractionInputs<int32_t> const&>(*inputs);
-                    validateTyped(typedReference, typedResult);
-                }
-                else if(m_problem.a().dataType() == DataType::BFloat16
-                     && m_problem.b().dataType() == DataType::BFloat16
-                     && m_problem.c().dataType() == DataType::BFloat16
-                     && m_problem.d().dataType() == DataType::BFloat16)
-                {
-                    auto const& typedReference = dynamic_cast<BFloat16ContractionInputs const&>(*m_referenceInputs);
-                    auto const& typedResult = dynamic_cast<BFloat16ContractionInputs const&>(*inputs);
-                    validateTyped(typedReference, typedResult);
-                }
-                else
-                {
-                    throw std::runtime_error("Data type not implemented.");
-                }
+              validateSolution(inputs);
```

Comment:
Not using the return value here. This is why I don't really like return codes to report errors.

Reply:
tricky case here,  with new code we are calling validateSolution from two places and want to record the comparison status in separate member fields.  Return value seems reasonable as opposed to making the validateSolution aware of the caller so it can set appropriate member var.